baseband/vdif/base.py

```diff
@@ -124,7 +124,8 @@ def find_header(self, template_header=None, framesize=None, edv=None,
                 if template_header.same_stream(header):
                     self.seek(-header.size, 1)
                     return header
-            # Didn't work, so get searching.
+            # Didn't work, so reset the file pointer and get searching.
+            self.seek(-template_header.size, 1)
```

Comment:
There's a more elegant solution - I just moved file_pos = self.tell() to the top of the function (since it's now being used in all cases regardless of the try/except outcome).

I also added a few comments on what the code is doing.  I don't know what the optimal amount of comments should be in a code, and would welcome your input on whether it's extraneous or not.